src/VectorizeLoops.cpp

```diff
@@ -994,6 +1240,169 @@ class VectorSubs : public IRMutator {
         : var(std::move(v)), replacement(std::move(r)), target(t), in_hexagon(in_hexagon) {
         widening_suffix = ".x" + std::to_string(replacement.type().lanes());
     }
+};  // namespace
+
+class FindVectorizableExprsInAtomicNode : public IRMutator {
+    Scope<> poisoned_names;
```

Comment:
I probably know what this means but might be good to comment about the definition of "poisoned".